src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -731,6 +731,15 @@ private void InitializeOutputFile(TOptions analyzeOptions, TContext context)
 
                         _run = new Run();
 
+                        if (!string.IsNullOrWhiteSpace(analyzeOptions.AutomationId) || !string.IsNullOrWhiteSpace(analyzeOptions.AutomationGuid))
+                        {
+                            _run.AutomationDetails = new RunAutomationDetails
+                            {
+                                Id = analyzeOptions.AutomationId,
+                                Guid = analyzeOptions.AutomationGuid
+                            };
```

Comment:
Use this. Make sure this doesn't emit empty values if these arguments aren't provided;

_run = new Run()
 {
    AutomationDetails = new RunAutomationDetails
    {
        Id = analyzeOptions.AutomationId,
        Guid = analyzeOptions.AutomationGuid
  }
};

Reply:
done!